src/helpers/ContractSynchronizer.ts

```diff
@@ -5,15 +5,64 @@ import getOwnedERC721, {
   parseLogData,
 } from 'helpers/getOwnedERC721'
 
+export interface ContractSynchronizerSchema {
+  account: string
+  synchronizedBlockId: number
+  addressToTokenIds: { [address: string]: string[] }
+}
+
 export default class ContractSynchronizer {
   account: string
   locked = false
   synchronizedBlockId?: number
   addressToTokenIds: { [address: string]: Set<string> } = {}
   skipTransactions = new Set<string>()
 
-  constructor(account: string) {
+  constructor(
+    account: string,
+    addressToTokenIds: { [address: string]: Set<string> } = {},
+    synchronizedBlockId?: number
+  ) {
     this.account = account
+    this.synchronizedBlockId = synchronizedBlockId
+    this.addressToTokenIds = addressToTokenIds
+  }
+
+  static fromJSON({
+    account,
+    synchronizedBlockId,
+    addressToTokenIds,
+  }: {
+    account: string
+    synchronizedBlockId: number
+    addressToTokenIds: { [address: string]: string[] }
+  }) {
+    const addressToTokenSetIds = Object.entries(addressToTokenIds).reduce(
```

Comment:
It's pretty hard to understand on what is going on here, maybe you can simplify it? Maybe we can extract it into a function?

Reply:
Added `transformObjectValues` ​​function :)